velox/exec/IndexLookupJoin.cpp

```diff
@@ -31,7 +31,94 @@ void duplicateJoinKeyCheck(
   }
   VELOX_USER_CHECK_EQ(lookupKeyNames.size(), keys.size());
 }
+
+std::string getColumnName(const core::TypedExprPtr& typeExpr) {
+  const auto field = core::TypedExprs::asFieldAccess(typeExpr);
+  VELOX_USER_CHECK(field->isInputColumn());
+  return field->name();
+}
+
+void addLookupInputColumn(
+    const std::string& columnName,
+    const TypePtr& columnType,
+    column_index_t columnChannel,
+    std::vector<std::string>& lookupInputNames,
+    std::vector<TypePtr>& lookupInputTypes,
+    std::vector<column_index_t>& lookupInputChannels,
+    folly::F14FastSet<std::string>& lookupInputNameSet) {
+  if (lookupInputNameSet.count(columnName) != 0) {
+    return;
+  }
+  lookupInputNames.emplace_back(columnName);
+  lookupInputTypes.emplace_back(columnType);
+  lookupInputChannels.emplace_back(columnChannel);
+  lookupInputNameSet.insert(columnName);
+}
+
+bool addBetweenConditionBound(
+    const core::TypedExprPtr& typeExpr,
+    const RowTypePtr& inputType,
+    const TypePtr& indexKeyType,
+    std::vector<std::string>& lookupInputNames,
+    std::vector<TypePtr>& lookupInputTypes,
+    std::vector<column_index_t>& lookupInputChannels,
+    folly::F14FastSet<std::string>& lookupInputNameSet) {
+  const bool isConstant = core::TypedExprs::isConstant(typeExpr);
+  if (!isConstant) {
+    const auto conditionColumnName = getColumnName(typeExpr);
+    const auto conditionColumnChannel =
+        inputType->getChildIdx(conditionColumnName);
+    const auto conditionColumnType = inputType->childAt(conditionColumnChannel);
+    VELOX_USER_CHECK(conditionColumnType->equivalent(*indexKeyType));
+    addLookupInputColumn(
+        conditionColumnName,
+        conditionColumnType,
+        conditionColumnChannel,
+        lookupInputNames,
+        lookupInputTypes,
+        lookupInputChannels,
+        lookupInputNameSet);
+  } else {
+    VELOX_USER_CHECK(core::TypedExprs::asConstant(typeExpr)->type()->equivalent(
+        *indexKeyType));
+  }
+  return isConstant;
+}
+
+void addBetweenCondition(
```

Comment:
Would you ad comments to explain what these functions do?

Reply:
Comment added.